src/mesh/mesh.cpp

```diff
@@ -1204,6 +1204,22 @@ int Mesh::GetNumberOfMeshBlockCells() const {
 }
 const RegionSize &Mesh::GetBlockSize() const { return base_block_size; }
 
+const IndexShape &Mesh::GetLeafBlockCellBounds(CellLevel level) const {
+  // TODO(JMM): Luke this is for your Metadata::fine stuff.
+  PARTHENON_DEBUG_REQUIRE(level != CellLevel::fine,
+                          "Currently no access to finer cellbounds");
+  MeshBlock *pmb = block_list[0].get();
```

Comment:
What are the implicit assumptions/implications here always using the first block?

Reply:
That all blocks have the same shape, which is actually not necessarily true in the multigrid case, but in that case there are separate block lists, so this one still is safe for that assumption.

Reply:
Also that's why this is called `GetLeafBlockCellBounds`.